x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/textembedding/TextEmbeddingMlNodeServiceTests.java

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ *
+ * This file was contributed to by a Generative AI
+ */
+
+package org.elasticsearch.xpack.inference.services.textembedding;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.inference.InferenceServiceExtension;
+import org.elasticsearch.inference.Model;
+import org.elasticsearch.inference.ModelConfigurations;
+import org.elasticsearch.inference.TaskType;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.inference.services.settings.MlNodeServiceSettings;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+
+import static org.hamcrest.Matchers.instanceOf;
+import static org.mockito.Mockito.mock;
+
+public class TextEmbeddingMlNodeServiceTests extends ESTestCase {
```

Comment:
I could't add tests for the customEland case because it requires uploading a model using the eland python client. I did perform manual testing for the customEland client when it was on a separate branch, the log of which you can find in the attached file. I will also be performing this manual testing on this whole PR once I finalize it.
[manual testing for custom eland models.txt](https://github.com/elastic/elasticsearch/files/14170941/manual.testing.for.custom.eland.models.txt)